src/site/content/en/blog/stick-to-compositor-only-properties-and-manage-layer-count/index.md

```diff
@@ -0,0 +1,109 @@
+layout: post
+title: Stick to Compositor-Only Properties and Manage Layer Count
+subhead: |
+  Compositing is where the painted parts of the page are put together for displaying on screen.
+authors:
+  - paullewis
+date: 2015-03-20
+updated: 2018-08-17 
+description: |
+  Compositing is where the painted parts of the page are put together for displaying on screen.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+
+---
+
+Compositing is where the painted parts of the page are put together for 
+displaying on screen.
+
+There are two key factors in this area that affect page performance: the number of compositor layers that need to be managed, and the properties that you use for animations.
+
+### TL;DR
+
+* Stick to transform and opacity changes for your animations.
+* Promote moving elements with `will-change` or `translateZ`.
+* Avoid overusing promotion rules; layers require memory and management.
+
+## Use transform and opacity changes for animations
+
+The best-performing version of the pixel pipeline avoids both layout and paint, and only requires compositing changes:
+
+<figure>
+{% Img src="image/T4FyVKpzu4WKF1kBNvXepbi08t52/4to9zPKpICVNFCPl7qwV.jpg", alt="The pixel pipeline with no layout or paint.", width="800", height="122" %}
+</figure>
+
+In order to achieve this you will need to stick to changing properties that can be handled by the compositor alone. Today there are only two properties for which that is true - `transform`s and `opacity`:
+
+<figure>
+{% Img src="image/T4FyVKpzu4WKF1kBNvXepbi08t52/VoLzvR2akAnQfKFVyyM9.jpg", alt="The properties you can animate without triggering layout or paint.", width="800", height="600" %}
+</figure>
+
+The caveat for the use of `transform`s and `opacity` is that the element on which you change these properties should be on _its own compositor layer_. In order to make a layer you must promote the element, which we will cover next.
+
+Note: If you’re concerned that you may not be able to limit your animations to just those properties, take a look at the [FLIP principle](https://aerotwist.com/blog/flip-your-animations), which may help you remap animations to changes in transforms and opacity from more expensive properties.
+
+## Promote elements that you plan to animate
+
+As we mentioned in the “[Simplify paint complexity and reduce paint areas](https://developers.google.com/web/fundamentals/performance/rendering/simplify-paint-complexity-and-reduce-paint-areas)” section, you should promote elements that you plan to animate (within reason, don’t overdo it!) to their own layer:
+
+```css
+.moving-element {
+    will-change: transform;
+}
+```
+
+Or, for older browsers, or those that don’t support will-change:
+
+```css
+.moving-element {
+    transform: translateZ(0);
+}
+```
+
+
+This gives the browser the forewarning that changes are incoming and, depending on what you plan to change, the browser can potentially make provisions, such as creating compositor layers.
```

Comment:
```suggestion
```

This gives the browser the forewarning that changes are incoming and, depending on what you plan to change, the browser can potentially make provisions, such as creating compositor layers.
```